secboot/secboot_tpm.go

```diff
@@ -445,14 +451,33 @@ func buildLoadSequences(pathSequences [][]string) []*sb.EFIImageLoadEvent {
 
 	loadEvents := make([]*sb.EFIImageLoadEvent, 0, len(pathSequences))
 
+	efiImage := func(name string) (sb.EFIImage, error) {
+		if strings.HasSuffix(name, ".snap") {
```

Comment:
my preference would be to be explicit about this and have something like:

```
type EFIImage interface  {  // to use in the params
   image()  sb.EFIImage
}

type FileEFIImage string

// ... implement image

type SnapEFIImage struct {
    Snap string
    Relative string
}

// ... implement image

```

Reply:
Thanks. Will do something to make it not hardcoded in a follow-up.